pkg/bucketaccessrequest/bucketaccessrequest.go

```diff
@@ -0,0 +1,141 @@
+package bucketaccessrequest
+
+import (
+	"context"
+
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kubernetes-sigs/container-object-storage-interface-api/apis/objectstorage.k8s.io/v1alpha1"
+	bucketclientset "github.com/kubernetes-sigs/container-object-storage-interface-api/clientset"
+	bucketcontroller "github.com/kubernetes-sigs/container-object-storage-interface-api/controller"
+	"github.com/kubernetes-sigs/container-object-storage-interface-controller/pkg/util"
+	kubeclientset "k8s.io/client-go/kubernetes"
+
+	"github.com/golang/glog"
+)
+
+type bucketAccessRequestListener struct {
+	kubeClient   kubeclientset.Interface
+	bucketClient bucketclientset.Interface
+}
+
+func NewListener() bucketcontroller.BucketAccessRequestListener {
+	return &bucketAccessRequestListener{}
+}
+
+func (b *bucketAccessRequestListener) InitializeKubeClient(k kubeclientset.Interface) {
+	b.kubeClient = k
+}
+
+func (b *bucketAccessRequestListener) InitializeBucketClient(bc bucketclientset.Interface) {
+	b.bucketClient = bc
+}
+
+func (b *bucketAccessRequestListener) Add(ctx context.Context, obj *v1alpha1.BucketAccessRequest) error {
+	glog.V(1).Infof("add called for bucketaccessrequest %s", obj.Name)
+	bucketAccessRequest := obj
+
+	err := b.provisionBucketAccess(ctx, bucketAccessRequest)
+	if err != nil {
+		// Provisioning is 100% finished / not in progress.
+		switch err {
+		case util.ErrInvalidBucketAccessClass:
+			glog.V(5).Infof("Bucket access class specified does not exist. Stop provisioning, removing bucket access request %s from bucket access requests in progress", bucketAccessRequest.UID)
```

Comment:
1) is this interpretation of `err` always correct? could this err also be that a field is missing in the BAC?
2) "Stop provisioning, removing bucket access request %s from bucket access requests in progress". This msg makes me think that the controller will next "stop provisioning", which is not correct.

What if each msg just interprets the returned err and at the end append the msg about provisioning stopped?  Eg.
```
switch err {
		case util.ErrInvalidBucketAccessClass:
			errmsg = sprintf("Bucket access class error: %v", err)
                case util.ErrBucketAccessAlreadyExists:
                        errmsg = sprintf("Bucket access already exist: %v", err)
                 ...
}
err = nil
errmsg = sprintf("%s\nProvisioning stopped.", errmsg)
```

Reply:
There are some errors that are not recoverable. There is no point in keeping the BAR in the queue.